packages/guides-restructured-text/src/RestructuredText/TextRoles/DefaultTextRoleFactory.php

```diff
@@ -17,19 +17,37 @@ class DefaultTextRoleFactory implements TextRoleFactory
      */
     public function __construct(
         private readonly TextRole $genericTextRole,
+        private TextRole $defaultTextRole,
         iterable $textRoles = [],
         private array $domains = [],
     ) {
         $this->textRoles = [...$textRoles];
     }
 
-    public function registerTextRole(TextRole $textRoles): void
+    public function registerTextRole(TextRole $textRole): void
     {
-        $this->textRoles[] = $textRoles;
+        $this->textRoles[] = $textRole;
```

Comment:
Should we not create an name based map to store the text roles? As this is there unique identifier?

Reply:
One testrole can have multiple aliases